src/pages/workspace/WorkspaceNewRoomPage.js

```diff
@@ -149,9 +206,9 @@ class WorkspaceNewRoomPage extends React.Component {
                             prefixCharacter="#"
                             placeholder={this.props.translate('newRoomPage.social')}
                             containerStyles={[styles.mb5]}
-                            onChangeText={roomName => this.setState({roomName: this.checkAndModifyRoomName(roomName)})}
+                            onChangeText={roomName => this.clearErrorAndSetValue('roomName', this.checkAndModifyRoomName(roomName).modifiedRoomName)}
```

Comment:
This will change, we should only be using the room name setter method
```suggestion
                            onChangeText={roomName => this.clearErrorAndSetValue('roomName', this.modifyRoomName(roomName))}
```